Zend/zend_vm_def.h

```diff
@@ -8768,6 +8768,91 @@ ZEND_VM_HANDLER(200, ZEND_FETCH_GLOBALS, UNUSED, UNUSED)
 	ZEND_VM_NEXT_OPCODE();
 }
 
+ZEND_VM_HANDLER(202, ZEND_SEND_PLACEHOLDER, UNUSED, UNUSED)
+{
+	USE_OPLINE
+	zval *arg;
+	zend_execute_data *call = EX(call);
+
+	arg = ZEND_CALL_ARG(call, opline->op2.num);
+
+	Z_TYPE_INFO_P(arg) = opline->op1.num;
+
+	if (Z_TYPE_INFO_P(arg) == _IS_PLACEHOLDER_VARIADIC) {
+		ZEND_ADD_CALL_FLAG(call, ZEND_CALL_VARIADIC_PLACEHOLDER);
```

Comment:
The handling of variadic placeholders seems a bit odd to me. You push a `PLACEHOLDER_VARIADIC` on the stack and set the `CALL_VARIADIC_PLACEHOLDER` runtime flag. The `PLACEHOLDER_VARIADIC` placeholder might get overridden by an explicit argument in zend_handle_named_arg, so I don't think it can really influence behavior, right?

As you know whether variadic placeholder is passed at compile-time, wouldn't it be better to just set an extended_value flag on DO_FCALL_PARTIAL?

Reply:
Yeah, I'll cleanup here also.